src/BloomExe/Publish/PdfViewer.cs

```diff
@@ -99,9 +99,13 @@ public bool ShowPdf(string pdfFile)
 				// and continue to show it using that.
 			}
 #endif
-
-			var url = string.Format("{0}{1}?file=/bloom/{2}", Bloom.web.ServerBase.PathEndingInSlash,
-				FileLocator.GetFileDistributedWithApplication("pdf/web/viewer.html"), pdfFile);
+			// Escaping the filename twice for characters like # is needed in order to get the
+			// pdf filename reliably through to our local server.  This is to prevent the filename
+			// from being cut short at the # character.  See https://jira.sil.org/browse/BL-951.
+			var url = string.Format("{0}{1}?file=/bloom/{2}",
+				Bloom.web.ServerBase.PathEndingInSlash,
+				FileLocator.GetFileDistributedWithApplication("pdf/web/viewer.html"),
+				pdfFile.EscapeCharsForHttp().EscapeCharsForHttp());
```

Comment:
It seems that if you escape it twice, something must unescape it twice...otherwise the end result will be to convert % to an escape sequence. But I can't find any new unescaping.

I'm also not really clear, even from your comment, why the double escaping is needed. I see code below that re-assembles the parts before and after the (first?) #.